pallets/ah-migrator/src/lib.rs

```diff
@@ -542,6 +599,48 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Pallet<T> {
+		/// Execute a stage transition and log it.
+		fn transition(new: MigrationStage) {
+			let old = AhMigrationStage::<T>::get();
+			AhMigrationStage::<T>::put(&new);
+			log::info!(
+				target: LOG_TARGET,
+				"[Block {:?}] Stage transition: {:?} -> {:?}",
+				frame_system::Pallet::<T>::block_number(),
+				&old,
+				&new
+			);
+			Self::deposit_event(Event::StageTransition { old, new });
```

Comment:
Maybe this should send an UMP each time there is a stage transition? Then the relay can keep track of progress and we only have to send it from one spot in the code.

Reply:
I would first see the results of dry run. It would be more clear what we need, and do we need optimize on migration time for example.

Reply:
But what we achieve by that? AH does not really know much about the stage (it does not know if all pallets data is sent). 
I can see that we can have AH sending an XCM message after it processes every XCM message, but it might increase the migration time significantly. 
Or we can let AH know that this message with the data is the last on this pallet, but AH still can get congested just on one pallet with a lot of data.

Reply:
Unclear yet on what we need exactly, seems fine.